src/core/temporal/qgsabstracttemporal.h

```diff
@@ -0,0 +1,125 @@
+/***************************************************************************
+                         qgsabstracttemporal.h
+                         ---------------
+    begin                : January 2020
+    copyright            : (C) 2020 by Samweli Mwakisambwe
+    email                : samweli at kartoza dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+
+#ifndef QGSABSTRACTTEMPORAL_H
+#define QGSABSTRACTTEMPORAL_H
+
+#include "qgis_core.h"
+#include "qgis_sip.h"
+#include "qgslogger.h"
+#include "qgsmessagelog.h"
+#include "qgsapplication.h"
+#include "qgsrange.h"
+
+#include <QObject>
+#include <QDateTime>
+
+/**
+ * \class QgsAbstractTemporal
+ * \ingroup core
+ * Base class for temporal based classes.
+ *
+ * Subclasses may wish to update the abstract temporal functions.
+ *
+ * \since QGIS 3.14
+ */
+class CORE_EXPORT QgsAbstractTemporal
+{
+  public:
+
+    /**
+     * Constructors for QgsAbstractTemporal.
+     */
+    QgsAbstractTemporal();
+
+    QgsAbstractTemporal( const bool enabled );
+
+    virtual ~QgsAbstractTemporal() = default;
+
+    /**
+     * Sets object as a temporal based one, which will be considered when rendering maps with a specific time range set.
+     *
+     * \see isTemporal()
+     * \since QGIS 3.14
```

Comment:
QGIS documentation convention is that we don't include \since directives for members if which duplicate the class level \since directive. (i.e. you'd only use \since for new members for this class if its "\since QGIS 3.16" or later)